### ISSUE TITLE
Fix regression where report flags don't exit with non-zero

### ISSUE THREAD

**Issue by XhmikosR** [stylelint/stylelint#5046]
> Clearly describe the bug

When a needless disable is found, the process exits with a zero code.

> What stylelint configuration is needed to reproduce the bug?

```json
{
  "extends": [
    "stylelint-config-twbs-bootstrap/scss"
  ],
  "rules": {
    "function-disallowed-list": [
      "calc",
      "lighten",
      "darken"
    ],
    "property-disallowed-list": [
      "border-radius",
      "border-top-left-radius",
      "border-top-right-radius",
      "border-bottom-right-radius",
      "border-bottom-left-radius",
      "transition"
    ],
    "scss/dollar-variable-default": [
      true,
      {
        "ignore": "local"
      }
    ],
    "scss/selector-no-union-class-name": true
  }
}
```

> Which version of stylelint are you using?

13.8.0

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

CLI: `stylelint "**/*.{css,scss}" --cache --cache-location .cache/.stylelintcache --rd`

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

Probably not?

> What did you expect to happen?

The process should error out and exit with non-zero code

> What actually happened (e.g. what warnings or errors did you get)?

The needless disables are printed, but the process exits with 0.

```shell
scss/_reboot.scss
 1:1  ×  Needless disable for "at-rule-no-vendor-prefix"  --report-needless-disables
```

---

Test repo:

1. `git clone https://github.com/twbs/bootstrap.git`
2. `npm i && npm run css-lint-stylelint`

---

Notes:

1. Running the same command one more time, does not print the needless disables at all unless I delete the cache file.
2. Those needless disables weren't caught before 13.8.0, I think. But when a needless disable was found, the process exited with non-zero code in the previous Stylelint version. So, it looks like this is a regression in 13.8.0.

**Comment by jeddy3** [comment]
@XhmikosR Thanks for the clear report and for using the template.

Yes, it appears to be a regression as the:
- CLI should exit with non-zero code
- Node.js API should set the `errored` property to `true`

When any of the `report*` flags report a violation.

> Running the same command one more time, does not print the needless disables at all unless I delete the cache file.

That doesn't sound good! We should create another issue to investigate that.

---

Ref: https://github.com/stylelint/stylelint/issues/4896#issuecomment-730596160

**Comment by XhmikosR** [comment]
> That doesn't sound good! We should create another issue to investigate that.

I'm not sure if it's a separate issue (although it sounds like it), I'll need to look into it more unless someone beats me to it :)

**Comment by jeddy3** [comment]
Possible fix detailed in https://github.com/stylelint/stylelint/issues/4896#issuecomment-731140804.

**Comment by edg2s** [comment]
When I originally reported this, I was expecting warnings from the Node API, as that is what the equivalent feature in ESLint does: https://eslint.org/docs/user-guide/configuring#report-unused-eslint-disable-comments

When moving the feature to the Node API they made this a warning to allow more rule fixes to be considered patch releases: https://github.com/eslint/eslint/issues/12703#issuecomment-568582014
